### PR TITLE
Implementing To Do List Project with HTML, CSS, and JavaScript

### PR DESCRIPTION
**Pull request by Peter-Kibet**
In this pull request, I have implemented the To Do List project using HTML, CSS, and JavaScript, following the given requirements and best practices. The project is set up with webpack and served by a webpack dev server. Below is an overview of the changes made:

General Requirements:
- Ensured there are no linter errors by following the coding conventions and rules.
- Followed GitHub Flow for version control and collaboration.
- Documented the work in a professional manner, providing clear comments and explanations in the code.

HTML/CSS Requirements:
- Followed the best practices for HTML and CSS, ensuring clean and well-structured code.

JavaScript Requirements:
- Followed the best practices for JavaScript, including code organization, modularity, and readability.

Project Requirements:
- Set up a new project with webpack based on the previously completed webpack exercise.
- Created the index.html file as the template, with a placeholder for the To Do List.
- Implemented the index.js file, containing an array of to-do tasks.
- Created a function to iterate over the tasks array and dynamically populate the HTML list with tasks.
- Rendered the list of tasks in the designated placeholder on page load, maintaining the order based on the index values of each task.
- Developed a style.css file to apply the necessary styling for the To Do List, loaded via Webpack Style/CSS Loader.
- Ensured that all source files (index.html, index.js, and style.css) are located in the /src directory, and the distribution files are generated by webpack and served by webpack dev server from the /dist folder.

Please review this pull request to incorporate the implemented To Do List project.

### CODE REVIEW

**Review by tresorsawasawa — COMMENTED** [review]
# STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @Peter-Kibet ,:wave: :wave: 
While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🛑 🛑

You have submitted a project that doesn't have a README for your project documentation. Make sure to use the required **[template](https://github.com/microverseinc/readme-template)** 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

---
Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

---
_Invalid Code Review Request **Does not count** into the code reviews limit._

File: src/index.js [hunk 1]
```diff
@@ -1,28 +1,98 @@
...
+const addTaskButton = document.getElementById('add-task-button');
+const newTaskInput = document.getElementById('new-task-input');
+const clearAllButton = document.getElementById('clear-all-button');
 
-  btn.innerHTML = "Click me and check the console!";
-  btn.onclick = printMe;
+const task1 = {
+  description: 'Complete JavaScript project',
+  completed: false,
+};
```
Comment (Efiamotu-1) [inline src/index.js:13]: - [ ] As shown in the image below detailing the projects requirement, each task is expected to have three keys (description, completed, index), you currently have two (description, completed). please i kindly suggest you add the third key which is `index` for every task. see the example below for how it should be done
- > project requirement
miss
- > example code 
```javascript
{
    description: 'An example description in string',
    completed: false,
    index: 1,
  },

```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+Coded with HTML, CSS and JavaScript.
+
+<!-- Features -->
+
+### Key Features <a name="key-features"></a>
+
+- [Feature_1]: Task Creation: Users can create new tasks by providing a title or description for the task.
+- [Feature_2]: Task Organization: Tasks can be organized into categories or projects, allowing users to group related tasks together.
+
+## Future Features <a name="future-features"></a>
```
Comment (Efiamotu-1) [inline README.md:37]: - [ ] This section is empty please add some contents to it so your readme looks professional :+1: :sparkles:

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @Peter-Kibet , 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by usorfaitheloho — APPROVED** [review]
Hi @Peter-Kibet  , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**